### ISSUE TITLE
Support type assignment checking

### ISSUE THREAD

**Issue by MeirionHughes** [MeirionHughes/aurelia-template-lint#136]
* [ ] For known types, check that a binding target is assignable from the access-chain target. 

Don't know how to do type assignment checking with the typescript library. I don't really want to create a program object ( to get a type-checker object) unless absolutely necessary. 

Awaiting feedback: http://stackoverflow.com/questions/40073701/create-a-typescript-typechecker-without-using-ts-createprogram

**Comment by MeirionHughes** [comment]
This might help. http://blog.scottlogic.com/2015/01/20/typescript-compiler-api.html 
Looks like you can implement your own host then make a language service; that then might allow getting the Program and then onto the typechecker.

**Comment by MeirionHughes** [comment]
I don't think TypeChecker will help. the "isTypeAssignableTo" stuff is all hidden away (checker.ts). :/

**Comment by atsu85** [comment]
Have You asked help from TypeScript team/project? The need for the workaround seems very odd.

**Comment by MeirionHughes** [comment]
No, got told off last time I asked a question on there. So far [the question on SO on this ](http://stackoverflow.com/questions/25014160/typescript-compiler-as-a-service-how-to-test-if-one-type-is-assignable-to-anoth)doesn't have a (good) answer.
